exporters/committime/app.py

```diff
@@ -22,233 +18,31 @@
-                logging.debug("sha: %s, commit_timestamp not found in cache, executing API call." % (commit_sha))
-                metric.getCommitTime()
-                # If commit time is None, then we could not get the value from the API
-                # This is due to calling a non Github API.
-                if metric.commit_time is None:
-                    return None
-                # Add the timestamp to the cache
-                self.commit_dict[metric.commit_hash] = metric.commit_timestamp
-            else:
-                metric.commit_timestamp = self.commit_dict[commit_sha]
-                logging.debug("Returning sha: %s, commit_timestamp: %s, from cache." % (
-                    commit_sha, metric.commit_timestamp))
-
-            return metric
-
-        except Exception as e:
-            logging.warning("Build %s/%s in app %s is missing required attributes to collect data. Skipping."
-                            % (namespace, build.metadata.name, app))
-            logging.debug(e, exc_info=True)
-            return None
-
-
-class CommitTimeMetric:
-    _prefix_pattern = "https://%s/repos/"
-    _defaultapi = "api.github.com"
-    _prefix = _prefix_pattern % _defaultapi
-    _suffix = "/commits/"
-
-    def __init__(self, app_name, _gitapi=None):
-        self.name = app_name
-        self.labels = None
-        self.repo_url = None
-        self.commiter = None
-        self.commit_hash = None
-        self.commit_time = None
-        self.commit_timestamp = None
-        self.build_name = None
-        self.build_config_name = None
-        self.image_location = None
-        self.image_name = None
-        self.image_tag = None
-        self.image_hash = None
-        if _gitapi is not None and len(_gitapi) > 0:
-            logging.info("Using non-default GitHub API: %s" % (_gitapi))
-            self._prefix = self._prefix_pattern % _gitapi
-
-    def getCommitTime(self):
-        myurl = self.repo_url
-        url_tokens = myurl.split("/")
-        url = self._prefix + url_tokens[3] + "/" + url_tokens[4].split(".")[0] + self._suffix + self.commit_hash
-        response = requests.get(url, auth=(username, token))
-        if response.status_code != 200:
-            # This will occur when trying to make an API call to non-Github
-            logging.warning("Unable to retrieve commit time for build: %s, hash: %s, url: %s. Got http code: %s" % (
-                self.build_name, self.commit_hash, url_tokens[2], str(response.status_code)))
-        else:
-            commit = response.json()
-            try:
-                self.commit_time = commit['commit']['committer']['date']
-                self.commit_timestamp = pelorus.convert_date_time_to_timestamp(self.commit_time)
-            except Exception:
-                logging.error("Failed processing commit time for build %s" % self.build_name, exc_info=True)
-                logging.debug(commit)
-                raise
+class GitFactory:
```

Comment:
we should default to `git_type=github` so that we don't break existing deployments. Right now if I rebuild an existing commit exporter, the pod will fail to start with:

```
Traceback (most recent call last):
  File "committime/app.py", line 45, in <module>
    REGISTRY.register(collector)
  File "/opt/app-root/lib/python3.6/site-packages/prometheus_client/registry.py", line 26, in register
    names = self._get_names(collector)
  File "/opt/app-root/lib/python3.6/site-packages/prometheus_client/registry.py", line 53, in _get_names
    desc_func = collector.collect
AttributeError: 'NoneType' object has no attribute 'collect'
```